malsim/scenario.py

```diff
@@ -331,9 +331,15 @@ def create_simulator_from_scenario(
         **kwargs
     )
 
+    # This version only supports one defender and one attacker
     for agent_id, agent_info in conf['agents'].items():
         if agent_info['type'] == "attacker":
-            sim.register_attacker(agent_id, 0)
+            all_attackers = list(attack_graph.attackers.values())
+            assert len(all_attackers) == 1, (
+                "You have defined more than one attacker,",
+                "can not decide which one belongs to agent in simulator"
```

Comment:
"cannot" and maybe make this a log error instead. To me I'd leave this as a warning and just still take the first attacker, but it's not a meaningful discussion anyway. So, just go with your gut.